include/lbann/base.hpp

```diff
@@ -44,6 +44,10 @@ namespace lbann {
 // Forward-declaration.
 class lbann_comm;
 
+// support for encodind data_id in conduit::Node, used by
+// conduit_data_store and associated code
+#define SAMPLE_ID_PAD 9
```

Comment:
Could we make this a `constexpr` variable instead? Or prefix it with `LBANN_`? I worry about a rather generic macro name conflicting with other stuff.